Analytics configuration is the responsibility of the host application when running in widget mode

hughns opened the pull request:
Part of https://github.com/element-hq/element-call/issues/2994. Best reviewed [without whitespaces](https://github.com/element-hq/element-call/pull/3089/files?w=1) changes.

When running in widget mode the host application (e.g. messenger application) is now responsible for managing analytics consent.

- The following configuration options are ignored when running in widget mode:
  - `sentry.*`
  - `posthog.*`
- New embedded package-only URL parameters:
  - `posthogApiHost`, `posthogApiKey`
  - `sentryDsn` and `sentryEnvironment`
  - `rageshakeSubmitUrl`
- `analyticsID` is deprecated in favour of `posthogUserId`
- The Analytics section of the Settings|Feedback is not shown in embedded package
- The Settings|Feedback tab is not visible at all in embedded package if `rageshakeSubmitUrl` is not specified

hughns committed:
Support for analytics configuration via URL parameters in widget mode

Adds:

- posthogApiHost
- posthogApiKey
- rageshakeSubmitUrl
- sentryDsn
- sentryEnvironment

Deprecate analyticsId and use posthogUserId instead

hughns commented on docs/url-params.md:
```diff
@@ -36,39 +36,45 @@ There are two formats for Element Call urls.
+| `preload`            | `true` or `false`                                                                         | No, defaults to `false`             | Pauses app before joining a call until an `io.element.join` widget action is seen, allowing preloading.                                                                                                                                              ...
+| `rageshakeSubmitUrl` | Rageshake server URL endpoint                                                             | No                                  | e.g. `https://element.io/bugreports/submit`                                                                                                                                                                                                          ...
+| `returnToLobby`      | `true` or `false`                                                                         | No, defaults to `false`             | Displays the lobby in widget mode after leaving a call; shows a blank page if set to `false`. Useful for video rooms.                                                                                                                                ...
+| `sentryDsn`          | Sentry [DSN](https://docs.sentry.io/concepts/key-terms/dsn-explainer/)                    | No, defaults to the value in config |                                                                                                                                                                                                                                                      ...
```
```suggestion
| `sentryDsn`          | Sentry [DSN](https://docs.sentry.io/concepts/key-terms/dsn-explainer/)                    | No | n.b. the value set in config is never used when running as widget                                                                                                                                                                                                                                                                                                                                                   |
```

hughns commented on docs/url-params.md:
```diff
@@ -36,39 +36,45 @@ There are two formats for Element Call urls.
+| `rageshakeSubmitUrl` | Rageshake server URL endpoint                                                             | No                                  | e.g. `https://element.io/bugreports/submit`                                                                                                                                                                                                          ...
+| `returnToLobby`      | `true` or `false`                                                                         | No, defaults to `false`             | Displays the lobby in widget mode after leaving a call; shows a blank page if set to `false`. Useful for video rooms.                                                                                                                                ...
+| `sentryDsn`          | Sentry [DSN](https://docs.sentry.io/concepts/key-terms/dsn-explainer/)                    | No, defaults to the value in config |                                                                                                                                                                                                                                                      ...
+| `sentryEnvironment`  | Sentry [environment](https://docs.sentry.io/concepts/key-terms/key-terms/)                | No, defaults to the value in config |                                                                                                                                                                                                                                                      ...
```
```suggestion
| `sentryEnvironment`  | Sentry [environment](https://docs.sentry.io/concepts/key-terms/key-terms/)                | No | n.b. the value set in config is never used when running as widget                                                                                                                                                                                                                                                                                                                                                   |
```

hughns commented on docs/url-params.md:
```diff
@@ -36,39 +36,45 @@ There are two formats for Element Call urls.
+| `baseUrl`            |                                                                                           | Yes                                 | The base URL of the homeserver to use for media lookups.                                                                                                                                                                                             ...
+| `deviceId`           | Matrix device ID                                                                          | Yes                                 | The Matrix device ID for the widget host.                                                                                                                                                                                                            ...
+| `parentUrl`          |                                                                                           | Yes                                 | The url used to send widget action postMessages. This should be the domain of the client or the webview the widget is hosted in. (in case the widget is not in an Iframe but in a dedicated webview we send the postMessages same WebView the widget ...
+| `posthogApiHost`     | Posthog server URL                                                                        | No                                  | e.g. `https://posthog-element-call.element.io`                                                                                                                                                                                                       ...
```
```suggestion
| `posthogApiHost`     | Posthog server URL                                                                        | No                                  | e.g. `https://posthog-element-call.element.io`. n.b. the value set in config is never used when running as widget                                                                                                                                                                                                                                                                                                     |
```

hughns commented on docs/url-params.md:
```diff
@@ -36,39 +36,45 @@ There are two formats for Element Call urls.
+| `deviceId`           | Matrix device ID                                                                          | Yes                                 | The Matrix device ID for the widget host.                                                                                                                                                                                                            ...
+| `parentUrl`          |                                                                                           | Yes                                 | The url used to send widget action postMessages. This should be the domain of the client or the webview the widget is hosted in. (in case the widget is not in an Iframe but in a dedicated webview we send the postMessages same WebView the widget ...
+| `posthogApiHost`     | Posthog server URL                                                                        | No                                  | e.g. `https://posthog-element-call.element.io`                                                                                                                                                                                                       ...
+| `posthogApiKey`      | Posthog project API key                                                                   | No                                  |                                                                                                                                                                                                                                                      ...
```
```suggestion
| `posthogApiKey`      | Posthog project API key                                                                   | No                                  | n.b. the value set in config is never used when running as widget                                                                                                                                                                                                                                                                                                                                                   |
```

BillCarsonFr reviewed (changes requested):
Is it possible to have some tests to assert the behavior?

hughns committed:
Partial test coverage

hughns committed:
Simplify tests

hughns committed:
More tests

hughns committed:
Lint

hughns commented:
> Is it possible to have some tests to assert the behavior?

The behaviour wasn't tested previously. I've added tests that cover the new functionality.

n.b. the opentelemetry isn't actually wired up at the moment, but I added some tests for it anyhow.

hughns committed:
Merge branch 'livekit' into hughns/embedded-extra-url-config

hughns committed:
Split embedded only parameters into own section for clarity

hughns commented on docs/url-params.md:
```diff
@@ -34,51 +34,64 @@ possible to support encryption.
+
+### Embedded-only parameters
+
+These parameters are only available in the [embedded](./embedded-standalone.md) package of Element Call and will be ignored in the [full](./embedded-standalone.md) package.
```
```suggestion
These parameters are only supported in the [embedded](./embedded-standalone.md) package of Element Call and will be ignored in the [full](./embedded-standalone.md) package.
```

hughns commented on docs/url-params.md:
```diff
@@ -34,51 +34,64 @@ possible to support encryption.
 ### Widget-only parameters
 
-These parameters are only available in widget mode.
+These parameters are only available in [widget](./embedded-standalone.md) mode.
```
```suggestion
These parameters are only supported in [widget](./embedded-standalone.md) mode.
```

robintown reviewed (approved):
Makes sense

hughns committed:
Update docs/url-params.md

hughns committed:
Update docs/url-params.md

hughns commented on vite.config.js:
```diff
@@ -16,6 +16,7 @@ import basicSsl from "@vitejs/plugin-basic-ssl";
 // https://vitejs.dev/config/
 export default defineConfig(({ mode, packageType }) => {
   const env = loadEnv(mode, process.cwd());
+  process.env.VITE_PACKAGE = packageType ?? "full";
```
```suggestion
  // Environment variables with the VITE_ prefix are accessible at runtime.
  // So, we set this to allow for build/package specific behaviour.
  // In future we might be able to do what is needed via code splitting at
  // build time.
  process.env.VITE_PACKAGE = packageType ?? "full";
```

hughns committed:
Update vite.config.js

BillCarsonFr reviewed (approved):
Thanks for adding all these tests 👍
As far as I can tell it looks ok.
